File: domain/src/main/java/com/meals/domain/dataSource/GetMealUseCase.kt
package com.meals.domain.dataSource

import com.meals.domain.base.ParamsUseCase
import com.meals.domain.model.Meal
import com.meals.domain.repository.MealRepository
import io.reactivex.Single

class GetMealUseCase(
    private val mealRepository: MealRepository
) : ParamsUseCase<GetMealUseCase.Params, Single<Meal>>() {

    override fun buildUseCaseObservable(params: Params): Single<Meal> =
        mealRepository.getMeal(params.id, params.code, params.date)

    data class Params(
        val id: String,
        val code: String,
        val date: String
    )
}

File: School_Food_iOS_V2/School_Food_iOS_V2/Domain/UseCase/GetMealUseCase.swift
//
//  GetMealUseCase.swift
//  School_Food_iOS_V2
//
//  Created by 박진 on 2020/12/01.
//

import Foundation
import RxSwift
import RxCocoa

class GetMealUseCase: ParamUseCase {
   
    
    let mealRepository: MealRepository
    
    init() {
        self.mealRepository = MealRepositoryImpl()
    }
    
    typealias Params = GetMealRequest
    typealias T = Single<Meal>
    
    func buildUseCaseObservable(params: GetMealRequest) -> Single<Meal> {
        return mealRepository.getMeal(date: params.date)
    }
}
